internal/metrics/middleware.go

```diff
@@ -0,0 +1,53 @@
+package metrics
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/lbryio/lbrytv/internal/errors"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type key int
+
+const timerContextKey key = iota
+
+// Measure middleware starts a timer whenever a request is performed.
+// It should be added as first in the chain of middlewares.
+// Note that it doesn't catch any metrics by itself,
+// HTTP handlers are expected to add their own by calling AddObserver.
+func Measure() mux.MiddlewareFunc {
```

Comment:
lets call this `Middleware()` to be consistent with the other middleware we have